ospfd/ospf_vty.c

```diff
@@ -5862,126 +5985,229 @@ static const char *const link_data_desc[] = {
 
 /* Show router-LSA each Link information. */
 static void show_ip_ospf_database_router_links(struct vty *vty,
-					       struct router_lsa *rl)
+					       struct router_lsa *rl,
+					       json_object *json)
 {
 	int len, type;
 	unsigned int i;
+	json_object *json_links;
+	json_object *json_link;
+	int metric = 0;
+
+	if (json)
+		json_links = json_object_new_object();
 
 	len = ntohs(rl->header.length) - 4;
 	for (i = 0; i < ntohs(rl->links) && len > 0; len -= 12, i++) {
 		type = rl->link[i].type;
 
-		vty_out(vty, "    Link connected to: %s\n",
-			link_type_desc[type]);
-		vty_out(vty, "     (Link ID) %s: %s\n", link_id_desc[type],
-			inet_ntoa(rl->link[i].link_id));
-		vty_out(vty, "     (Link Data) %s: %s\n", link_data_desc[type],
-			inet_ntoa(rl->link[i].link_data));
-		vty_out(vty, "      Number of TOS metrics: 0\n");
-		vty_out(vty, "       TOS 0 Metric: %d\n",
-			ntohs(rl->link[i].metric));
-		vty_out(vty, "\n");
+		if (json) {
+			char link[10];
+
+			snprintf(link, 10, "Link-%d", i);
+			json_link = json_object_new_object();
+			json_object_string_add(json_link, "linkType",
+					       link_type_desc[type]);
+			json_object_string_add(json_link, link_id_desc[type],
+					       inet_ntoa(rl->link[i].link_id));
+			json_object_string_add(
+				json_link, link_data_desc[type],
+				inet_ntoa(rl->link[i].link_data));
+			json_object_int_add(json_link, "numOfTOSmetrics",
+					    metric);
+			json_object_int_add(json_link, "TOS_0_Metric",
```

Comment:
Change it to tos0Metric. In camel-casing, we don't use underscore.

Reply:
Done